### ISSUE TITLE
Remove/update references to NamedData

### ISSUE THREAD

**Issue by bangerth** [dealii/dealii#641]
The `NamedData` has this comment, right at the top of the class documentation:
```
/**
 * @deprecated The use of this class is deprecated and AnyData should be used
 * instead. It is not only more flexible, the way constness was assured in
 * this class never worked the way it was intended. Therefore, the according
 * checks have been disabled.
```
However, it is used in a great many places that aren't themselves deprecated:
```
include/deal.II/algorithms/newton.templates.h:  Newton<VECTOR>::operator() (NamedData<VECTOR *> &out, const NamedData<VECTOR *> &in)
include/deal.II/algorithms/newton.templates.h:        NamedData<VECTOR *> out;
include/deal.II/algorithms/newton.templates.h:            NamedData<VECTOR *> out;
include/deal.II/algorithms/newton.h:   * When calling (*#residual)(), the NamedData <tt>in</tt> given to the
include/deal.II/algorithms/newton.h:    virtual void operator() (NamedData<VECTOR *> &out, const NamedData<VECTOR *> &in);
include/deal.II/algorithms/operator.templates.h:    NamedData<VECTOR *> new_out;
include/deal.II/algorithms/operator.templates.h:          deallog << "Cannot convert AnyData argument " << out.name(i) << " to NamedData"
include/deal.II/algorithms/operator.templates.h:    NamedData<VECTOR *> new_in;
include/deal.II/algorithms/operator.templates.h:            // in NamedData. As soon as NamedData is gone, this code
include/deal.II/algorithms/operator.templates.h:            // in NamedData. As soon as NamedData is gone, this code
include/deal.II/algorithms/operator.templates.h:                  << " to NamedData" << std::endl;
include/deal.II/algorithms/operator.templates.h:  Operator<VECTOR>::operator() (NamedData<VECTOR *> &out, const NamedData<VECTOR *> &in)
include/deal.II/algorithms/operator.templates.h:  OutputOperator<VECTOR>::operator<< (const NamedData<VECTOR *> &vectors)
include/deal.II/algorithms/operator.h:   * NamedData to access these vectors. Typically, the first vector in
include/deal.II/algorithms/operator.h:    virtual void operator() (NamedData<VECTOR *> &out, const NamedData<VECTOR *> &in);
include/deal.II/algorithms/operator.h:   * An unary operator base class, intended to output the vectors in NamedData
include/deal.II/algorithms/operator.h:     * @deprecated Output all the vectors in NamedData.
include/deal.II/algorithms/operator.h:    OutputOperator<VECTOR> &operator<< (const NamedData<VECTOR *> &vectors);
include/deal.II/algorithms/any_data.h: * @todo Deprecate access by index after NamedData has been deprecated for
include/deal.II/algorithms/any_data.h:   * @brief Merge the data of another NamedData to the end of this object.
include/deal.II/algorithms/any_data.h:  /// Conversion from old NamedData
include/deal.II/algorithms/any_data.h:  AnyData(const NamedData<type> &);
include/deal.II/algorithms/any_data.h:   * name, but NamedData had a different name in that position.
include/deal.II/algorithms/any_data.h:AnyData::AnyData(const NamedData<type> &other)
```

@guidokanschat : What's the long term plan for this class? Should we formally attach `DEAL_II_DEPRECATED` to the class and all of the member functions that use it? Or what is your preference?

**Comment by guidokanschat** [comment]
The NamedData in Newton's method is for compatibility. I will look into it
when I have done my reviewing duties.

**Comment by bangerth** [comment]
I believe that `NamedData` has been removed as a class, but there are still a few occurrences of it in comments:
```
1/dealii> grep -r NamedData include/ source/ examples/
include/deal.II/algorithms/newton.h:   * When calling (*#residual)(), the NamedData <tt>in</tt> given to the
include/deal.II/algorithms/operator.h:   * NamedData to access these vectors. Typically, the first vector in
include/deal.II/algorithms/named_selection.h: * Select data from NamedData corresponding to the attached name.
include/deal.II/algorithms/named_selection.h:   * Add a new name to be searched for in NamedData.
include/deal.II/algorithms/named_selection.h:   * Return the corresponding index in the NamedData object supplied
include/deal.II/algorithms/any_data.h: * @todo Deprecate access by index after NamedData has been deprecated for
include/deal.II/algorithms/any_data.h:   * name, but NamedData had a different name in that position.
```

@guidokanschat : Can these all be removed or replaced?

**Comment by bangerth** [comment]
@guidokanschat ?

**Comment by bangerth** [comment]
@guidokanschat ?

**Comment by tjhei** [comment]
fixed by #2005
